File: app/src/main/java/org/dhis2/data/schedulers/SchedulersModule.java
package org.dhis2.data.schedulers;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class SchedulersModule {

    @Provides
    @Singleton
    SchedulerProvider schedulerProvider() {
        return new SchedulersProviderImpl();
    }
}

File: app/src/main/java/org/dhis2/data/schedulers/SchedulersModule.kt
package org.dhis2.data.schedulers

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

@Module
class SchedulersModule {

    @Provides
    @Singleton
    internal fun schedulerProvider(): SchedulerProvider {
        return SchedulersProviderImpl()
    }
}
